infrastructure/src/main/java/com/masegosadev/infrastructure/persistence/videogame/VideogameCollection.java

```diff
@@ -27,11 +26,26 @@ public VideogameCollection(InMemoryVideogameRepository inMemoryVideogameReposito
 
     @Override
     public Videogame createVideogame(Videogame videogame) {
-        Set<PlatformEntity> platformEntities = videogame.getPlatforms().stream().map(namePlatformToPlatformEntity).collect(Collectors.toSet());
-        VideogameEntity videoGameEntity = new VideogameEntity(videogame.getName() + "::VIDEOGAME", videogame.getName(), videogame.getImageUrl(), platformEntities);
-        VideogameEntity savedVideogame = inMemoryVideogameRepository.save(videoGameEntity);
-        Set<String> platforms = savedVideogame.getPlatformEntities().stream().map(platformEntityToNamePlatform).collect(Collectors.toSet());
+        VideogameEntity videogameSaved = saveVideoGame(videogame);
+        return returnVideoGame(videogameSaved);
+    }
+
+    private VideogameEntity saveVideoGame(Videogame videogame) {
+        Set<PlatformEntity> platformEntities = getPlatforms(videogame.getPlatforms());
+        VideogameEntity videogameEntity = new VideogameEntity(videogame.getName() + "::VIDEOGAME", videogame.getName(), videogame.getImageUrl(), platformEntities);
+        return inMemoryVideogameRepository.save(videogameEntity);
+    }
+
+    private Videogame returnVideoGame(VideogameEntity videogameEntity) {
+        Set<String> platforms = getNamePlatforms(videogameEntity.getPlatformEntities());
+        return new Videogame(videogameEntity.getVideoGameName(), videogameEntity.getImageUrl(), platforms);
```

Comment:
You can use as parameter videogame entity in getNamePlatforms, creating a shorter method signature and in that way you can use ir as parameter of Videogame